src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -40,6 +58,136 @@ abstract class MethodFingerprint(
+                append(accessFlags)
+                append(returnTypeValue.first())
+                if (parameters != null) appendSignatureKeyParameters(parameters)
+            }
+            return signatureMap[key]!!
+        }
+
+        internal fun clearMethodLookupMap() {
+            allMethods.clear()
+            signatureMap.clear()
+            stringMap.clear()
+        }
+
+        /**
+         * resolve faster, by creating culled lists based on method signature and Strings contained.
+         */
+        internal fun createMethodLookupMap(classes: Iterable<ClassDef>) {
+            fun addMethodToMapList(map: MutableMap<String, MutableList<ClassAndMethod>>,
+                                   key: String, keyValue:  ClassAndMethod) {
+                var list = map[key]
+                if (list == null) {
+                    list = LinkedList()
+                    map[key] = list
+                }
+                list += keyValue
+            }
+
+            for (classDef in classes) {
+                for (method in classDef.methods) {
+                    // Key structure is: (access)(returnType)(optional: parameter types)
+                    val accessFlagsReturnKey = method.accessFlags.toString() + method.returnType.first()
+                    val accessFlagsReturnParametersKey = buildString {
+                        append(accessFlagsReturnKey)
+                        appendSignatureKeyParameters(method.parameterTypes)
+                    }
+                    val classAndMethod = ClassAndMethod(classDef, method)
+
+                    // For signatures with no access or return type specified.
+                    allMethods += classAndMethod
+                    // Access and return type.
+                    addMethodToMapList(signatureMap, accessFlagsReturnKey, classAndMethod)
+                    // Access, return, and parameters.
+                    addMethodToMapList(signatureMap, accessFlagsReturnParametersKey, classAndMethod)
+
+                    // Look up by Strings (the fastest way to resolve).
+                    method.implementation?.instructions?.forEach { instruction ->
+                        if (instruction.opcode == Opcode.CONST_STRING || instruction.opcode == Opcode.CONST_STRING_JUMBO) {
+                            val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+                            addMethodToMapList(stringMap, string, classAndMethod)
+                        }
+                    }
+
+                    // The only additional lookup that could benefit, is a map of the full class name to its methods.
+                    // This would require adding a 'class name' field to MethodFingerprint,
+                    // as currently the class name can be specified only with a custom fingerprint.
+                }
+            }
+        }
+
+        /**
+         * Resolve using map built in [createMethodLookupMap]
```

Comment:
Comments should end with punctuation and be properly worded:

```suggestion
         * Resolve using the lookup map built by [createMethodLookupMap].
```